core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1093,18 +1093,23 @@ protected Scope visitTableExecute(TableExecute node, Optional<Scope> scope)
 
             TableExecuteHandle executeHandle =
                     metadata.getTableHandleForExecute(
-                                    session,
-                                    tableHandle,
-                                    procedureName,
-                                    tableProperties)
+                            session,
+                            tableHandle,
+                            procedureName,
+                            tableProperties,
+                            procedureMetadata.getExecutionMode().isReadsData())
```

Comment:
passing `procedureMetadata.getExecutionMode().isReadsData()` here feels wrong. 
We are going to `ConnectorMetadata` via `MetadaManager.getTableHandleForExecute`. And `ConnectorMetadata` knows (should know) if given procedure reads or does not read data.

We can either make `MetadaManager.getTableHandleForExecute` call out to `TableProceduresRegistry` to obtain value of `readsData` flag for given procedure.

OR

We can drop `readsData` flag from `TableProcedureMetadata` at all and pass it to `TableExecuteHandle` via `ConnectorTableExecuteHandle`.

I am opting for the latter.

Reply:
continued at https://github.com/trinodb/trino/pull/10810#discussion_r796573452